util/test_singleplayer.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+# Runs a singleplayer session with software-rendering.
+
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+gameid=${gameid:-devtest}
+executable=$dir/../bin/luanti
+testspath=$dir/../tests
+conf_client=$testspath/client.conf
+worldpath=$testspath/world
+
+[ -e "$executable" ] || { echo "executable $executable missing"; exit 1; }
+
+rm -rf "$worldpath"
+mkdir -p "$worldpath/worldmods"
+
+# enable a lot of visual effects so we can catch shader errors and other obvious bugs
+printf '%s\n' >"$conf_client" \
+	screen_w=384 screen_h=256 fps_max=5 \
```

Comment:
Will this pick the default driver? (I.e. opengl)

NM: It's picked beforehand...